lib/versioning/ruby/range.ts

```diff
@@ -18,14 +18,14 @@ const parse = (range: string): Range => {
   const value = (range || '').trim();
 
   const match = regExp.exec(value);
-  if (match) {
-    const { version = null, operator = null, delimiter = ' ' } = match.groups;
+  if (match?.groups) {
+    const { version = '', operator = '', delimiter = ' ' } = match.groups;
     return { version, operator, delimiter };
   }
 
   return {
-    version: null,
-    operator: null,
+    version: '',
+    operator: '',
     delimiter: ' ',
   };
 };
```

Comment:
I think a new test case should be added so that we know how to handle the case where a new value cannot be calculated and we have to figure out what to return in such case null or empty string. For now I have returned empty string as string is the return type of the function.

Reply:
Please check the calling function and the versioning API, maybe it should be widen to return null

Reply:
The result of the `parse` function is used in `isVersion` and `isSingleOperator` which do not take in `null` type as argument. Should I widen their types?
Relevant code:-
https://github.com/renovatebot/renovate/blob/bda9df5a00bbda8bdf2d1464c73ac64c981cc348/lib/versioning/ruby/index.ts#L50-L56

Reply:
> Needs test / lint fixes

New tests are not needed after the change I made, can you verify it please:
Old Code:-
```js
const lastPart: string = range
      .split(',')
      .map((part) => part.trim())
      .pop()
  ```
  New Code:-
  ```js
  const lastPart: string = range
      .split(',')
      .map((part) => part.trim())
      .slice(-1)
      .join();
  ```
  
 `pop` return undefined in case the array is empty, which causes null-check error down the line where we use `lastPart`. So I replaced it with `.slice(-1).join()` which return string in all cases.

Reply:
@viceice Can you review the changes...

Reply:
@viceice @rarkins Could you please review the changes..